ci/test-checks.sh

```diff
@@ -75,7 +75,7 @@ fi
 
  _ ci/order-crates-for-publishing.py
 
-nightly_clippy_allows=()
+nightly_clippy_allows=(--allow=clippy::redundant_clone)
```

Comment:
Why did you add this @ryoqun ?

Reply:
@CriesofCarrots in short, I was exhausted for bumping nightly this time, which took months. (details: https://github.com/solana-labs/solana/pull/31381#discussion_r1186698539)

just saw #31690. sorry for trouble because of inter-branch clippy inconsistency. to be honest, i didn't anticipated 
this.

after initial triage from @brooksprumo, i sensed fixing this isn't straigtforward: https://github.com/solana-labs/solana/pull/31381#discussion_r1186711957

so, i opted to merge this pr in as-is. (this urgency is coming from the fact my upcoming pr will be blocked on rustc 1.70+)

seems @apfitzge is starting to tackle on this: #31685.

initially, i thought this exception isn't end of world thing, but seems to cause actual annoyance. I'll work on this.

so, put differently, crude way to solicit collaboration. ;)